interp/src_test.go

```diff
@@ -222,7 +222,9 @@ func Test_previousRoot(t *testing.T) {
 		t.Run(test.desc, func(t *testing.T) {
 			t.Parallel()
 
-			p := previousRoot(test.root)
+			// TODO(mpl): add tests for new algo.
+			// it's not too bad to not have them for now, since the tests in example run this code path too.
+			p, _ := previousRoot("vendor", test.root)
```

Comment:
I do not get why you pass `"vendor"` here as it is defined  and used as a const in `previousRoot`.

You have also changed the `previousRoot` signature to return an error, but you are not testing or using it.

Reply:
Passing "vendor" as argument should guarantee that the new code is _not_ used for these tests. (Ah, if you mean that we could use a const instead of an hardcoded literal, then yes, sure. The const was not in my head, I added it because the linter forced me to). Again, I was pressed by time, and just wanted to be sure to keep these tests passing. Hence the TODO for later.

Reply:
done